kin-core/src/main/java/kin/core/ServiceProvider.java

```diff
@@ -87,8 +87,12 @@ KinAsset getKinAsset() {
 
         boolean isKinBalance(@NonNull AccountResponse.Balance balance) {
             return stellarKinAsset.getCode().equals(balance.getAssetCode()) &&
-                balance.getAssetIssuer() != null &&
-                stellarKinAsset.getIssuer().getAccountId().equals(balance.getAssetIssuer().getAccountId());
+                balance.getAsset() != null &&
+                isKinAsset(balance.getAsset());
+        }
+
+        boolean isKinAsset(@NonNull Asset asset) {
+            return stellarKinAsset.equals(asset);
```

Comment:
not sure I understand two different issuer can create the same asset name?

Reply:
correct

Reply:
so why did you remove tellarKinAsset.getIssuer().getAccountId().equals(balance.getAssetIssuer().getAccountId())?

Reply:
I wanted to share code, because I have the same check, during this, I've found out I missed a check (check the actual asset code) and that I can use equals, so I shared both new and old impl to this one.
I need to add to unit test this case that I've missed - the same issuer with a different asset code

Reply:
after additional look, I made a mistake, I did check it before, should have removed the entire check there, and replace by the equals, now fixed.